File: app/src/main/java/eu/sesma/dagger/di/ApplicationComponent.java
package eu.sesma.dagger.di;

import eu.sesma.dagger.platform.ApplicationModule;
import eu.sesma.dagger.ui.IAppCollaborator;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        ApplicationModule.class
})
public interface ApplicationComponent {

    //Exposed to sub-graphs
    Context provideContext();

    IAppCollaborator provideAppCollaborator();
}

File: app/src/main/java/eu/sesma/dagger/di/ApplicationComponent.kt
package eu.sesma.dagger.di

import android.app.Application
import dagger.Component
import eu.sesma.dagger.IAppSingletonCollaborator
import eu.sesma.dagger.core.di.CoreComponent

@ApplicationScoped
@Component(
        modules = [ApplicationModule::class],
        dependencies = [CoreComponent::class])
interface ApplicationComponent {

    //Exposed to sub-graphs
    fun provideContext(): Application

    fun appSingletonCollaborator(): IAppSingletonCollaborator

}
